File: treinos/treino003/app/src/main/java/aluno/gabriel/treino003/view/SplashActivity.java
package aluno.gabriel.treino003.view;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.TextView;

import aluno.gabriel.treino003.R;
import aluno.gabriel.treino003.core.ApplicationUtil;
import aluno.gabriel.treino003.model.Usuario;

public class SplashActivity extends AppCompatActivity {
    Usuario u1;
    Bundle bundle;
    TextView txtVersion;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        applicationVersion();
        trocarTela();
    }

    protected void trocarTela(){
        boolean Handler = new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                userDataSource();
                Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                intent.putExtras(bundle);
                startActivity(intent);
                finish();
            }
        },5000);
    }

    protected void applicationVersion(){
        txtVersion = findViewById(R.id.txtVersion);
        txtVersion.setText("Versão: "+ ApplicationUtil.APP_VERSION);
    }

    protected void userDataSource(){
        u1 = new Usuario(
                "Roger Santos Silva",
                "roger.santos14@hotmail.com",
                "4430esipK4@m"
        );
        bundle = new Bundle();
        bundle.putString("name", u1.getNome());
        bundle.putString("email", u1.getEmail());
        bundle.putString("password", u1.getSenha());
    }

}

File: treinos/treino014/app/src/main/java/aluno/gabriel/treino014/view/SplashActivity.kt
package aluno.gabriel.treino014.view

import aluno.gabriel.treino014.R
import aluno.gabriel.treino014.core.AppUtil
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.TextView

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        editFields()
        trocarTela()
    }

    fun trocarTela(){
        Handler(Looper.getMainLooper()).postDelayed({
            startActivity(Intent(this@SplashActivity, MainActivity::class.java))
            finish()
        },10000)
    }

    fun editFields(){
        val txtNome: TextView = findViewById(R.id.txtNome)
        val txtVersion: TextView = findViewById(R.id.txtVersion)
        txtVersion.text = "Versão: "+AppUtil.APP_VERSION
    }
}